### PR TITLE
Adapt learning rate to DART

### PR DESCRIPTION
**Pull request by guolinke**
also provide a dart version from xgboost

### CODE REVIEW

File: src/boosting/dart.hpp [hunk 1]
```diff
@@ -84,19 +86,16 @@ class DART: public GBDT {
...
-    // select dropping tree indexes based on drop_rate
-    // if drop rate is too small, skip this step, drop one tree randomly
-    if (gbdt_config_->drop_rate > kEpsilon) {
+    bool is_skip = random_for_drop_.NextDouble() < gbdt_config_->skip_drop;
+    // select dropping tree indexes based on drop_rate and tree weights
+    if (!is_skip) {
+      double inv_average_weight = static_cast<double>(tree_weight_.size()) / sum_weight_;
       for (int i = 0; i < iter_; ++i) {
-        if (random_for_drop_.NextDouble() < gbdt_config_->drop_rate) {
+        if (random_for_drop_.NextDouble() < gbdt_config_->drop_rate * tree_weight_[i] * inv_average_weight) {
```
Comment (wxchan) [inline src/boosting/dart.hpp]: 1. weighted drop is not default in xgboost, default is uniform
2. if `drop_rate * tree_weight_[i] * inv_average_weight` >1, but it will be calculated as 1, will it cause expectation of drop_rate smaller?

Comment (guolinke) [inline src/boosting/dart.hpp]: @wxchan 1. on my test, drop by weight has better accuracy. However, add uniform drop back is also ok.
2. I think this is not a big problem. if it >1, it means  ```tree_weight_[i]``` is very large, and is a small probability event.
